### PR TITLE
Buy Button Functionality Changes

### PR DESCRIPTION
**Pull request by kf-rns**
## Description
Removed unused variable from recipe url and changed buy functionality from Ok button to Buy 
Fixes #
- Remove unused variable from url
- Added buy functionality on buy button instead of modal

## Checklist
- [ ] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Run `meteor npm run lint`
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: imports/api/blocks/server/methods.js [hunk 1]
```diff
@@ -401,7 +401,7 @@ Meteor.methods({
                         validators[v].valconsAddress = Meteor.call('hexToBech32', validators[v].address, Meteor.settings.public.bech32PrefixConsAddr);
                         // validators[v].address = Meteor.call('getAddressFromPubkey', validators[v].pubKey);
                         // tempValidators[validators[v].pubKey.value] = validators[v];
-                        //validators[v].deeplink_url = 'http://wallet.pylons.tech/?action=purchase_nft&recipe_id=' + validators[v].valconsAddress + '&nft_amount=' + ntf_amount; //20210612
+                        //validators[v].deeplink_url = 'http://wallet.pylons.tech/?action=purchase_nft&recipe_id=' + validators[v].valconsAddress; //20210612
```
Comment (faisalnaveedRNS) [inline imports/api/blocks/server/methods.js]: Please remove dead code

File: imports/startup/server/index.js [hunk 2]
```diff
@@ -221,7 +221,7 @@ Meteor.startup(() => {
             }
             
         }  
-        else if (querys['?action'] == "resell_nft" && querys['recipe_id'] != null /*&& querys['cookbook_id'] != null*/ && querys['nft_amount'] == 1) { 
+        else if (querys['?action'] == "resell_nft" && querys['recipe_id'] != null /*&& querys['cookbook_id'] != null*/) {
```
Comment (faisalnaveedRNS) [inline imports/startup/server/index.js]: Please remove dead code

File: imports/ui/easel_buy/EaselBuy.jsx [hunk 3]
```diff
@@ -112,23 +110,11 @@ export default class EaselBuy extends Component {
...
-  }
-
-
   // In case IOS will redirect to APP Store if app not installed
   // In case Android will redirect to Play store if app not installed
   // In case in Browser will redirect to Play store
-  handleLoginConfirmed = (success) => {
-    if (success) {
+  handleLoginConfirmed = () => {
+    // if (success) {
```
Comment (faisalnaveedRNS) [inline imports/ui/easel_buy/EaselBuy.jsx]: remove dead code

File: imports/api/nfts/server/methods.js [hunk 4]
```diff
@@ -96,8 +96,7 @@ Meteor.methods({
                 "?action=resell_nft&recipe_id=" +
                 item.recipeID +
```
Comment (mustafapylons) [inline imports/api/nfts/server/methods.js]: We should sanitize how we are creating the URLs

File: imports/ui/easel_buy/EaselBuy.jsx [hunk 5]
```diff
@@ -24,225 +23,200 @@ new FlowRouterTitle(FlowRouter);
...
+                        for (let i = 0; i < strings.length; i++) {
+                            try {
+                                if (
+                                    (strings[i].key =
+                                        "NFT_URL" && strings[i].value.indexOf("http") >= 0)
+                                ) {
+                                    img = strings[i].value;
+                                    break;
+                                }
+                            } catch (e) {
```
Comment (mustafapylons) [inline imports/ui/easel_buy/EaselBuy.jsx:91]: better variable naming than "e" would be good.

**Review by mustafapylons — COMMENTED** [review]
Some comments, otherwise code is good.

**Review by BlackPanthar — COMMENTED** [review]
Reviewed. Will approve after addressing the other comments

**Review by BlackPanthar — APPROVED** [review]

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

File: imports/api/recipes/server/methods.js [hunk 6]
```diff
@@ -58,8 +58,7 @@ Meteor.methods({
             "?action=purchase_nft&recipe_id=" +
             recipe.ID +
             "&cookbook_id=" +
-            recipe.cookbookID +
-            "&nft_amount=1";
+            recipe.cookbookID;
```
Comment (MikeSofaer) [inline imports/api/recipes/server/methods.js:61]: Do we need this? Recipe already knows the cookbook it's in.

File: imports/ui/home/Home.jsx [hunk 7]
```diff
@@ -25,7 +25,6 @@ export default class Home extends Component {
     if (
       querys["?action"] == "purchase_nft" &&
```
Comment (MikeSofaer) [inline imports/ui/home/Home.jsx:26]: Would love to remove this

File: imports/ui/home/Home.jsx [hunk 8]
```diff
@@ -25,7 +25,6 @@ export default class Home extends Component {
...
       querys["recipe_id"] != null &&
-      querys["nft_amount"] == 1 &&
       querys["cookbook_id"] != null
```
Comment (MikeSofaer) [inline imports/ui/home/Home.jsx:28]: And this

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**